### ISSUE TITLE
Support Expo SDK 50+

### ISSUE THREAD

**Issue by marko-mihajlovic** [skam22/watermelondb-expo-plugin#25]
Seems like this plugin doesn't actually support expo 50 & 51, because of Java-Kotlin change. The plugin still uses Java [here](https://github.com/skam22/watermelondb-expo-plugin/blob/v51.0.2/plugin/src/index.ts#L104).

From [Expo SDK 50](https://expo.dev/changelog/2024/01-18-sdk-50) blog:  
![image](https://github.com/skam22/watermelondb-expo-plugin/assets/88029880/01b5b678-9a8c-4481-9058-93340a44ce40)
![image](https://github.com/skam22/watermelondb-expo-plugin/assets/88029880/73317f15-bcca-489b-b864-eedc06a1a682)

**Comment by skam22** [comment]
Good catch - if you want to send up a PR I’d be happy to merge it, otherwise I can go through it later. Thanks!

**Comment by marko-mihajlovic** [comment]
Unfortunately, RN 0.74 introduced [breaking change](https://reactnative.dev/blog/2024/04/22/release-0.74#other-breaking-changes) removing "[getJSIModulePackage](https://github.com/facebook/react-native/pull/42059/files#diff-de4b4eeeaab4a88ee88c172f745d4b344c41d4bc7ef46c9389e110f46d45cd85L130)". WatermelonDB needs to support new architecture for JSI to work, as requested in [this issue](https://github.com/Nozbe/WatermelonDB/issues/1769). After that, the plugin can be updated accordingly.

**Comment by LovesWorking** [comment]
> Unfortunately, RN 0.74 introduced [breaking change](https://reactnative.dev/blog/2024/04/22/release-0.74#other-breaking-changes) removing "[getJSIModulePackage](https://github.com/facebook/react-native/pull/42059/files#diff-de4b4eeeaab4a88ee88c172f745d4b344c41d4bc7ef46c9389e110f46d45cd85L130)". WatermelonDB needs to support new architecture for JSI to work, as requested in [this issue](https://github.com/Nozbe/WatermelonDB/issues/1769). After that, the plugin can be updated accordingly.

https://github.com/Nozbe/WatermelonDB/issues/1851#issuecomment-2507903366

it's supported now. Hopefully we can get this plugin updated.
